cuml/kmeans.py

```diff
@@ -0,0 +1,99 @@
+# Copyright (C) 2017-2019 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import argparse
+from bench import (
+    parse_args, time_mean_min, load_data, gen_basic_dict, output_csv
+)
+import numpy as np
+from cuml import KMeans
+
+parser = argparse.ArgumentParser(description='scikit-learn K-means benchmark')
+parser.add_argument('-i', '--filei', '--fileI', '--init',
+                    type=str, help='Initial clusters')
+parser.add_argument('-t', '--tol', type=float, default=0.,
+                    help='Absolute threshold')
+parser.add_argument('--maxiter', type=int, default=100,
+                    help='Maximum number of iterations')
+parser.add_argument('--samples-per-batch', type=int, default=32768,
+                    help='Maximum number of iterations')
+parser.add_argument('--n-clusters', type=int, help='Number of clusters')
+params = parse_args(parser, prefix='cuml', loop_types=('fit', 'predict'))
+
+# Load and convert generated data
+X_train, X_test, _, _ = load_data(params)
+
+if params.filei is not None:
+    X_init = np.load(params.filei).astype(params.dtype)
+    params.n_clusters = X_init.shape[0]
+else:
+    np.random.seed(params.seed)
```

Comment:
Comments were added.
Init algorithm might be added, but will give different centroids for cuML and DAAL. Since threshold is set to 0 by default, it will not affect training performance, but otherwise one of libraries possibly will perform lesser number of iterations.
So, I think that for this pull request this initialization method is enough.

Reply:
Ok, sounds good - thanks for investigating. Whenever I run this benchmark, I always import cluster centers from the file, so what you have here is fine for my usage.